python-spec/src/somacore/coordinates.py

```diff
@@ -0,0 +1,521 @@
+            )
+
+    def __matmul__(self, other: Any) -> CoordinateTransform:
+        if isinstance(other, CoordinateTransform):
+            if self.input_axes != other.output_axes:
+                raise ValueError("Axis mismatch between transformations.")
+            if isinstance(other, ScaleTransform):  # Includes IdentityTransform
+                return ScaleTransform(
+                    other.input_axes,
+                    self.output_axes,
+                    self.scale_factors * other.scale_factors,
+                )
+            if isinstance(other, AffineTransform):
+                return AffineTransform(
+                    other.input_axes,
+                    self.output_axes,
+                    self.augmented_matrix.__matmul__(other.augmented_matrix),
+                )
+        if isinstance(other, np.ndarray):
+            raise NotImplementedError(
+                "Support for multiplying by numpy arrays is not yet implemented."
+            )
+        raise TypeError(
+            f"Cannot multiply a CoordinateTransform by type {type(other)!r}."
+        )
+
+    def __rmatmul__(self, other: Any) -> CoordinateTransform:
+        if isinstance(other, CoordinateTransform):
+            if other.input_axes != self.output_axes:
+                raise ValueError("Axis mismatch between transformations.")
+            if isinstance(other, IdentityTransform):
+                return ScaleTransform(
+                    self.input_axes, other.output_axes, self._scale_factors
+                )
+            if isinstance(other, ScaleTransform):
+                return ScaleTransform(
+                    self.input_axes,
+                    other.output_axes,
+                    self._scale_factors * other._scale_factors,
+                )
+            if isinstance(other, AffineTransform):
+                return AffineTransform(
+                    self.input_axes,
+                    other.output_axes,
+                    other.augmented_matrix.__matmul__(self.augmented_matrix),
+                )
+        if isinstance(other, np.ndarray):
+            raise NotImplementedError(
+                "Support for multiplying by numpy arrays is not yet implemented."
+            )
+        raise TypeError(
+            f"Cannot multiply a CoordinateTransform by type {type(other)!r}."
+        )
+
+    @property
+    def augmented_matrix(self) -> npt.NDArray[np.float64]:
+        """Returns the augmented affine matrix for the transformation.
+
+        Lifecycle: experimental
+        """
+        scales = np.append(self.scale_factors, [1.0])
+        return np.diag(scales)
+
+    def inverse_transform(self) -> CoordinateTransform:
+        """Returns the inverse coordinate transform.
+
+        Lifecycle: experimental
+        """
+        return ScaleTransform(
+            self.output_axes, self.input_axes, 1.0 / self._scale_factors
+        )
+
+    @property
+    def isotropic(self) -> bool:
```

Comment:
I debated creating an `UniformScaleTransform` subclass for uniform scale transforms instead of adding it as a property. Thinking through your questions here convinced me that is more correct, since it's really what I'm trying to support anyway.